typescript/simple-mercari-web/src/components/ItemList/ItemList.tsx

```diff
@@ -17,27 +21,26 @@ interface Prop {
 
 export const ItemList: React.FC<Prop> = (props) => {
   const { reload = true, onLoadCompleted } = props;
-  const [items, setItems] = useState<Item[]>([])
+  const [items, setItems] = useState<Item[]>([]);
   const fetchItems = () => {
-    fetch(server.concat('/items'),
-      {
-        method: 'GET',
-        mode: 'cors',
-        headers: {
-          'Content-Type': 'application/json',
-          'Accept': 'application/json'
-        },
-      })
-      .then(response => response.json())
-      .then(data => {
-        console.log('GET success:', data);
+    fetch(server.concat("/items"), {
```

Comment:
`fetch`は`Promise`を返す非同期関数なので、呼び出す時に`await`をつける必要があります。なお、`await`は基本的に`async`がついた関数の中でしか呼び出せません。
```suggestion
    await fetch(server.concat("/items"), {
```
非同期関数(今回はfetch)に対して`await`を使わないと**その関数の処理が終わる前に次の処理が始まってしまいます**

今回のケースではあまり問題になっていませんが、より複雑なエラー処理をするときや、`fetch`の結果をもとに処理を行いたいときに不具合の原因になります。

Reply:
`then`と`catch`を使っているから処理してから次にいくと思ったのですが`await`を使わないと困ることがあるのでしょうか？
そもそも`then`, `catch`を使わず`await`を使ったほうがいいですか？

Reply:
あーーごめんなさい😭
ちゃんと後続処理もちゃんとチェーンで描いてるのでこのパターンでは不要でしたね。大変失礼しました🙇
また、今回のようにシンプルなパターンではthen/catchで全く問題ありません。(Listing.tsxの方も同様です🙇)

じゃあ同じことが出来るのになぜasync/awaitがあるかというと主に可読性や保守性の観点から選択することが多いです。
現実的なユースケースとして、通信後に
- snaqbarやloading barなどのUIのためのstateやglobalなstateを複数更新したい
- 受け取ったレスポンスからさらに細かく分岐して画面を遷移させたい
- エラーのstatusから文言を決定したい
- さらに別の非同期処理を呼び出したい

といった動作が増えた時、then/catchの中のネストがどんどん深くなっていく場合があります。
そういう時にasync/awaitを使ってかくと読みやすいコードになるため、コードが複雑になりやすい大きい開発だと好まれます。(特に非同期からさらに別の非同期は開発中頻繁にでくわします。)

Reply:
なるほどです！ありがとうございます:bow: